File: problems/2068-check-whether-two-strings-are-almost-equivalent/solution.py
class Solution:
    def checkAlmostEquivalent(self, word1: str, word2: str) -> bool:
        freq = {}
        for char in word1:
            freq[char] = freq.get(char, 0) + 1
        for char in word2:
            freq[char] = freq.get(char, 0) - 1
        for count in freq.values():
            if abs(count) > 3:
                return False
        return True

File: problems/2068-check-whether-two-strings-are-almost-equivalent/solution.c
bool checkAlmostEquivalent(char* word1, char* word2) {
    int freq[26] = {0};
    for (int i = 0; word1[i] != '\0'; i++) {
        freq[word1[i] - 'a']++;
    }
    for (int i = 0; word2[i] != '\0'; i++) {
        freq[word2[i] - 'a']--;
    }
    for (int i = 0; i < 26; i++) {
        if (freq[i] > 3 || freq[i] < -3) {
            return false;
        }
    }
    return true;
}